lib/Conversion/TorchOnnxToTorch/DefaultDomainQtoZ.cpp

```diff
@@ -914,6 +914,57 @@ void mlir::torch::onnx_c::populateDefaultDomainQtoZ(
                                         /*storeValue=*/data, keepDims,
                                         noop_with_empty_axes, false);
                 });
+  patterns.onOp(
+      "ReduceLogSumExp", 1,
+      [](OpBinder binder, ConversionPatternRewriter &rewriter) {
+        Torch::ValueTensorType resultType;
+        Value data;
+        int64_t keepDims, noop_with_empty_axes;
+        if (binder.tensorOperandAtIndex(data, 0) ||
+            binder.tensorResultType(resultType) ||
+            binder.s64IntegerAttr(keepDims, "keepdims", 1) ||
+            binder.s64IntegerAttr(noop_with_empty_axes, "noop_with_empty_axes",
+                                  0))
+          return failure();
+
+        // out = Log(reduce_sum(exp(data)))
+        Value castDType = rewriter.create<Torch::ConstantIntOp>(
+            binder.getLoc(), rewriter.getI64IntegerAttr(/*Float32Type*/ 11));
+        Value noneVal = rewriter.create<Torch::ConstantNoneOp>(binder.getLoc());
+        Value constFalse =
+            rewriter.create<Torch::ConstantBoolOp>(binder.getLoc(), false);
+        auto size = data.getType()
+                        .dyn_cast<Torch::ValueTensorType>()
+                        .getOptionalSizes();
+        auto f32ResultType = rewriter.getType<Torch::ValueTensorType>(
+            size, rewriter.getF32Type());
```

Comment:
Hi @archana-ramalingam, refer: https://github.com/llvm/torch-mlir/blob/6abc7371c8279a678a757af2fbf085502bd7e696/lib/Conversion/TorchOnnxToTorch/DefaultDomainAtoF.cpp#L58-L59

The value `11` means double dtype. So:
1.) The `f32ResultType` has to be changed to `f64ResultType`.
2.) The `castDType` value should contain the torch equivalent for that dtype, which in this case should be `7`.

You can refer: https://onnx.ai/onnx/operators/onnx__ReduceLogSumExp.html

Reply:
Do the /*Float32Type*/ and rewriter.getF32Type() need to be changed as well?

Reply:
Yeah, you have to make them f64.